python/cudf/cudf/_lib/join.pyx

```diff
@@ -171,10 +210,15 @@ cpdef join(Table lhs,
     all_cols_py = columns_from_ptr(move(c_result))
     if left_index or right_index:
         ix_odict = OrderedDict()
-        ix_odict[result_index_name] = all_cols_py.pop(result_index_pos)
-        index_col = Table(ix_odict)
+        for name, pos in zip(
+            result_index_names[::-1], result_idx_positions[::-1]
+        ):
+            ix_odict[name] = all_cols_py.pop(pos)
+        other_dict = OrderedDict()
+        for k, v in list(ix_odict.items())[::-1]:
```

Comment:
Is there a reason you need to convert the `odict_items` into a `list` here? It seems like an unnecessary copy.

Reply:
Not necessarily. Really I just want to reverse the order of iteration, because the items were previously popped backwards. I will look into some alternatives.

Reply:
Should be able to just do `reversed(ix_odict.items())`.

https://docs.python.org/3/library/functions.html#reversed